tests/testsuite/profile_trim_paths.rs

```diff
@@ -0,0 +1,517 @@
+    -Zremap-path-scope=object \
+    --remap-path-prefix=[..]/lib/rustlib/src/rust=/rustc/[..] \
+    --remap-path-prefix=[CWD]= [..]
+[FINISHED] dev [..]
+[RUNNING] `target/debug/foo[EXE]`"
+        ))
+        .run();
+}
+
+#[cargo_test(nightly, reason = "-Zremap-path-scope is unstable")]
+fn path_dependency_outside_workspace() {
+    let bar = project()
+        .at("bar")
+        .file("Cargo.toml", &basic_manifest("bar", "0.0.1"))
+        .file("src/lib.rs", r#"pub fn f() { println!("{}", file!()); }"#)
+        .build();
+    let bar_path = bar.url().to_file_path().unwrap();
+    let bar_path = bar_path.display();
+
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                [package]
+                name = "foo"
+                version = "0.0.1"
+
+                [dependencies]
+                bar = { path = "../bar" }
+
+                [profile.dev]
+                trim-paths = "object"
+           "#,
+        )
+        .file("src/main.rs", "fn main() { bar::f(); }")
+        .build();
+
+    p.cargo("run --verbose -Ztrim-paths")
+        .masquerade_as_nightly_cargo(&["-Ztrim-paths"])
+        .with_stdout("bar-0.0.1/src/lib.rs")
+        .with_stderr(&format!(
+            "\
+[COMPILING] bar v0.0.1 ([..]/bar)
+[RUNNING] `rustc [..]\
+    -Zremap-path-scope=object \
+    --remap-path-prefix=[..]/lib/rustlib/src/rust=/rustc/[..] \
+    --remap-path-prefix={bar_path}=bar-0.0.1 [..]
+[COMPILING] foo v0.0.1 ([CWD])
+[RUNNING] `rustc [..]\
+    -Zremap-path-scope=object \
+    --remap-path-prefix=[..]/lib/rustlib/src/rust=/rustc/[..] \
+    --remap-path-prefix=[CWD]= [..]
+[FINISHED] dev [..]
+[RUNNING] `target/debug/foo[EXE]`"
+        ))
+        .run();
+}
+
+#[cargo_test(nightly, reason = "-Zremap-path-scope is unstable")]
+fn diagnostics_works() {
+    Package::new("bar", "0.0.1")
+        .file("Cargo.toml", &basic_manifest("bar", "0.0.1"))
+        .file("src/lib.rs", r#"pub fn f() { let unused = 0; }"#)
+        .publish();
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                [package]
+                name = "foo"
+                version = "0.0.1"
+
+                [dependencies]
+                bar = "0.0.1"
+
+                [profile.dev]
+                trim-paths = "diagnostics"
+           "#,
+        )
+        .file("src/lib.rs", "")
+        .build();
+
+    let registry_src = paths::home().join(".cargo/registry/src");
+    let pkg_remap = format!("{}/[..]/bar-0.0.1=bar-0.0.1", registry_src.display());
+
+    p.cargo("build -vv -Ztrim-paths")
+        .masquerade_as_nightly_cargo(&["-Ztrim-paths"])
+        .with_stderr_contains(
+            "\
+warning: unused variable: `unused`
+ --> bar-0.0.1/src/lib.rs:1:18
+  |
+1 | pub fn f() { let unused = 0; }
+  |                  ^^^^^^ help: if this is intentional, prefix it with an underscore: `_unused`
+  |
+  = note: `#[warn(unused_variables)]` on by default",
```

Comment:
Tests can't match on the compiler's output exactly, since it changes too often, and makes it impossible to change (since it would require simultaneously changing both rust-lang/rust and rust-lang/cargo).

I think you can do something like `with_stderr_contains("[..]bar-0.0.1/src/lib.rs:1[..]")` to just check the important part.

Reply:
Thanks for calling this out, and replaced this with `with_stderr_line_without`.